### DISCUSSION TITLE
Doing inference of a custom trained yolov8 .pt model using pytorch

### DISCUSSION THREAD

**Discussion by swaminathanayyappan190799** [ultralytics/ultralytics#2426] [General]
Hi,

   Can someone help me to load my custom trained yolov8 object detection model using torch, actually i tried it using torch.hub.load("ultralytics/ultralytics", "custom", path="<pathofmymodel .pt>") method but it got failed due to the missing of hubconf.py, any suggestions to this issue is more welcomed ! 
   
![image](https://user-images.githubusercontent.com/47708135/236444683-bba0d035-8a29-46ae-819b-917e0caf8cb5.png)

**Comment by glenn-jocher** [comment]
@swaminathanayyappan190799 hello!

To load your custom trained YOLOv8 model using PyTorch, you don't need to use `torch.hub.load`. Instead, you can directly load your `.pt` file using PyTorch's `torch.load` function, followed by initializing the model and loading the weights.

Here's a general outline of the steps you would follow:

1. Import the necessary PyTorch modules.
2. Define the model architecture (this should match the architecture used during training).
3. Load your custom trained weights into the model using `torch.load`.

Make sure that the model architecture you define in your code matches the one used for training your `.pt` file. Once the model is loaded, you can perform inference by passing your images through the model.

For more detailed instructions on how to perform these steps, please refer to the Ultralytics Docs, specifically the sections related to the Predict mode and model loading. 😊

Good luck with your object detection tasks!
